language-support/ts/codegen/src/TsCodeGenMain.hs

```diff
@@ -388,12 +388,13 @@ genDefDataType curPkgId conName mod tpls def =
                             ] ++
                             ["};"]
                         associatedTypes =
-                          let tT = conName
-                              tK = tT <> ".Key"
+                          let mbKeyDef = fst . genType (moduleName mod) . tplKeyType <$> tplKey tpl
+                              tT = conName
+                              tK = maybe "undefined" (const (tT <> ".Key")) mbKeyDef
```

Comment:
Just out of curiosity, have you tried using just `Key` instead of `T.Key`? Did that work?

Reply:
Somebody seems to love the magic weapons for point-free code, `maybe` & `const`... 😃
I would have gone for
```suggestion
                              tK = if isJust mbKeyDef then tT <> ".Key" else "undefined"
```

Reply:
> Just out of curiosity, have you tried using just `Key` instead of `T.Key`? Did that work?

I have as a matter of fact. Yes, that works fine.